zilliqa/src/consensus.rs

```diff
@@ -1319,11 +1344,7 @@ impl Consensus {
         cosigned: BitVec,
     ) -> QuorumCertificate {
         // we've already verified the signatures upon receipt of the responses so there's no need to do it again
-        QuorumCertificate {
-            signature: NodeSignature::aggregate(signatures).unwrap(),
-            cosigned,
-            block_hash,
-        }
+        QuorumCertificate::new(signatures, cosigned, block_hash)
```

Comment:
It seems poor practice to put the signature aggregation outside of the QC constructor